### ISSUE TITLE
"await" as an ObjectBindingIdentifier in Formal Parameter for an async function not throwing error

### ISSUE THREAD

**Issue by Shilpi3** [shapesecurity/shift-parser-js#416]
The scripts 
```async ({await}) => 1` 
async function x({await}) { return 1 }
```
should throw an error in the parser. Instead, it is parsed as the following AST.
{
  "type": "Script",
  "directives": [],
  "statements": [
    {
      "type": "ExpressionStatement",
      "expression": {
        "type": "ArrowExpression",
        "isAsync": true,
        "params": {
          "type": "FormalParameters",
          "items": [
            {
              "type": "ObjectBinding",
              "properties": [
                {
                  "type": "BindingPropertyIdentifier",
                  "binding": {
                    "type": "BindingIdentifier",
                    "name": "await"
                  },
                  "init": null
                }
              ]
            }
          ],
          "rest": null
        },
        "body": {
          "type": "LiteralNumericExpression",
          "value": 1
        }
      }
    }
  ]
}
```

The script does not throw an early Error as well.

**Comment by bakkot** [comment]
It's also incorrectly allowed in the body of async functions:

```js
async function f() { return { await }; }
```
should be a parse error, but is not.

**Comment by bakkot** [comment]
Fixed by #418.
